% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/align_dia_runs.R
\name{alignTargetedRuns}
\alias{alignTargetedRuns}
\title{Outputs intensities for each analyte from aligned Targeted-MS runs}
\usage{
alignTargetedRuns(
  dataPath,
  outFile = "DIAlignR",
  params = paramsDIAlignR(),
  oswMerged = TRUE,
  scoreFile = NULL,
  runs = NULL,
  peps = NULL,
  refRun = NULL,
  applyFun = lapply
)
}
\arguments{
\item{dataPath}{(string) path to xics and osw directory.}

\item{outFile}{(string) name of the output file.}

\item{params}{(list) parameters are entered as list. Output of the \code{\link{paramsDIAlignR}} function.}

\item{oswMerged}{(logical) TRUE for experiment-wide FDR and FALSE for run-specific FDR by pyprophet.}

\item{scoreFile}{(string) path to the peptide score file, needed when oswMerged is FALSE.}

\item{runs}{(string) names of xics file without extension.}

\item{peps}{(integer) ids of peptides to be aligned. If NULL, align all peptides.}

\item{refRun}{(string) reference for alignment. If no run is provided, m-score is used to select reference run.}

\item{applyFun}{(function) value must be either lapply or BiocParallel::bplapply.}
}
\value{
An output table with following columns: precursor, run, intensity, RT, leftWidth, rightWidth,
 peak_group_rank, m_score, alignment_rank, peptide_id, sequence, charge, group_label.
}
\description{
This function expects osw and xics directories at dataPath. It first reads osw files and fetches chromatogram indices for each analyte.
It then align XICs of its reference XICs. Best peak, which has lowest m-score, about the aligned retention time is picked for quantification.
}
\examples{
params <- paramsDIAlignR()
params[["context"]] <- "experiment-wide"
dataPath <- system.file("extdata", package = "DIAlignR")
BiocParallel::register(BiocParallel::MulticoreParam(workers = 4, progressbar = TRUE))
alignTargetedRuns(dataPath, outFile = "testDIAlignR", params = params, applyFun = BiocParallel::bplapply)
}
\references{
Gupta S, Ahadi S, Zhou W, RÃ¶st H. "DIAlignR Provides Precise Retention Time Alignment Across Distant Runs in DIA and Targeted Proteomics." Mol Cell Proteomics. 2019 Apr;18(4):806-817. doi: https://doi.org/10.1074/mcp.TIR118.001132 Epub 2019 Jan 31.
}
\seealso{
\code{\link{getRunNames}, \link{getFeatures}, \link{setAlignmentRank}, \link{getMultipeptide}}
}
\author{
Shubham Gupta, \email{shubh.gupta@mail.utoronto.ca}

ORCID: 0000-0003-3500-8152

License: (c) Author (2019) + GPL-3
Date: 2019-12-14
}
